### PR TITLE
#133 and #132: Clarifications labels and axis and fixes surface

### PR DESCRIPTION
**Pull request by matthiaskoenig**
- clarifies the labels
- clarifies what to display in legends
- updates description of Surface in line with AbstractCurve (and describes deprecated attributes).
- typo fixes

### CODE REVIEW

File: specification/level-1-version-4/sources/output.tex [hunk 1]
```diff
@@ -28,7 +28,7 @@ \subsubsection{\element{Plot}}
...
 
 \paragraph*{\element{legend}}
-The \element{legend} attribute defines whether a legend should be displayed (\val{true}) or not (\val{false}).  The position and styling of the legend is unspecified.  If the attribute is not defined, it is up to the tool whether to display the legend or not, and does not mean that the attribute has a default value of \val{false}.
+The \element{legend} attribute defines whether a legend should be displayed (\val{true}) or not (\val{false}).  The position and styling of the legend is unspecified.  If the attribute is not defined, it is up to the tool whether to display the legend or not, and does not mean that the attribute has a default value of \val{false}.  The \element{name} of an \AbstractCurve in a \Plot2D and  the \element{name} of a \Surface in a \Plot3D are used as labels in the legend, if defined. If not the respective \element{id} should be used as label.
```
Comment (fbergmann) [inline specification/level-1-version-4/sources/output.tex:31]: i would rather use the name of the datagenerator before using the curve id

File: specification/level-1-version-4/sources/output.tex [hunk 2]
```diff
@@ -145,12 +144,13 @@ \subsubsection{\element{ShadedArea}}
...
+A \Surface is a three-dimensional \Output component representing a (processed) simulation result (\fig{surface}).  A \Surface is a parallel class to \AbstractCurve that defines a three-dimensional surface instead of a two-dimensional curve.  Zero or more \Surface instances define a \PlotThree (\fig{output}).
+
+In addition to the optional \element{id} and \element{name} attributes it inherits from \SedBase (the latter of which should be used as the label in the \Plot3D legend, if present), it also defines the attributes \element{xDataReference}, \element{yDataReference}, and \element{zDataReference}, all of type \SIdRef, the first two of which are optional and the last of which is required.  It also defines the optional attributes \element{style} of type \SIdRef, and \element{type}, of type \SurfaceType.
```
Comment (fbergmann) [inline specification/level-1-version-4/sources/output.tex:153]: This needs to be \PlotThree instead of \Plot3D

**Comment by fbergmann** [comment]
@matthiaskoenig i again, not sure about the 'name'  of the datagenerator would you now not use it at all? Also What again is a surface with only `z` values and no `x` and `y` ones?

**Comment by matthiaskoenig** [comment]
I would use the name on the `DataGenerator` or the name on the `DataSet` to have a Human readable description of what is described by the `DataGenerator`. I never would use the `name` on the `DataSet`. I want to have access to this description whereever I use the DataGenerator, i.e. in reports and curves/plots. With the `DataSet.name` I would be limited to reports and have to generate a report to store name the information.

Personally I think we should get rid of either `name` or `label` on `DataSet`.

Surface with only z values and no x and y is very similar to the curve without x data. It is just plotting the surface over the x-index and y-index (e.g. the `range(x_range)` and `range(y_range)` which was used to generate the 3D data). But without actual examples/implementations of the 3D surfaces we have no clue if this works so far.

File: specification/level-1-version-4/sources/output.tex (hunk 2 above)
Comment (matthiaskoenig) [reply specification/level-1-version-4/sources/output.tex]: fixed

**Comment by luciansmith** [comment]
Sorry for not noticing this earlier--I've now manually merged the changes here that hadn't made it into the document already.  I left off the sentence in the 'legend' section on the basis that it would take more words that this had to explain everything, but everything was already explained in the relevant curve/surface sections anyway.  Let me know if you disagree!

**Comment by matthiaskoenig** [comment]
Great, thanks. Closing this.

**Pull request closed without merging**